core/modules/replica/TestApp.cc

```diff
@@ -0,0 +1,133 @@
+} /// namespace
+
+
+namespace lsst {
+namespace qserv {
+namespace replica {
+
+TestApp::Ptr TestApp::create(int argc, char* argv[]) {
+    return Ptr(new TestApp(argc, argv));
+}
+
+
+TestApp::TestApp(int argc, char* argv[])
+    :   Application(
+            argc, argv,
+            "This is a simple demo illustrating how to use class Application"
+            " for constructing user applications with very little efforts spent"
+            " on typical tasks.",
+            true /* injectDatabaseOptions */,
+            true /* boostProtobufVersionCheck */,
+            true /* enableServiceProvider */
+        ),
+        _p2("ONE"),
+        _o1(123),
+        _o2(false) {
+
+    /* Configure the parser for the following syntax:
+     *
+     * COMMAND1 <p1> <p11> [<o1>] [<o11>] [--o12=<v>] [--verbose]
+     * COMMAND2 <p1>       [<o1>]                     [--verbose] [--f21]
+     * COMMAND2 <p1>       [<o1>]                     [--verbose]
+     * 
+     * Note that the parser is guaranteed to run before invoking method "runImpl()".
+     */
+    parser().commands(
+        "command",
+        {"COMMAND1", "COMMAND2", "COMMAND3"},
+        _cmd);
+
+    parser().required(
+        "p1",
+        "description of the required parameter p1 for all commands",
+        _p1);
+
+    parser().optional(
+        "o1",
+        "description of the optional parameter o1 for all commands",
+        _o1);
+
+    parser().flag(
+        "verbose",
+        "verbose mode",
+        _verbose);
+
+    auto&& command1 = parser().command("COMMAND1");
+    command1.description(
+        "This is the first command"
+    );
+    command1.required(
+        "p11",
+        "description of the additional required parameter specific for the command",
+        _p11
+    );
+    command1.optional(
+        "o11",
+        "description of the additional optional parameter specific for the command",
+        _o11
+    );
+    command1.option(
+        "o12",
+        "description of the additional option specific to the command",
+        _o12
+    );
+
+    parser().command("COMMAND2").description(
+        "This is the second command"
+    );
+    parser().command("COMMAND2").flag(
+       "f21",
+       "description of the additional flag specific to the command",
+       _f21
+    );
+}
+
+
+int TestApp::runImpl() {
+    cout
+        << "Hello from TestApplication:\n"
+        << "      cmd: " << _cmd << "\n"
+        << "      p11: " << _p11 << "\n"
+        << "       p1: " << _p1  << "\n"
+        << "       p2: " << _p2  << "\n"
+        << "       o1: " << _o1  << "\n"
+        << "       o2: " << (_o2      ? "true" : "false") << "\n"
+        << "      o11: " << _o11 << "\n"
+        << "      o12: " << _o12 << "\n"
+        << "      f21: " << (_f21     ? "true" : "false") << "\n"
+        << "  verbose: " << (_verbose ? "true" : "false") << endl;
+    return 0;
+}
+
+}}} // namespace lsst::qserv::replica
```

Comment:
If this is an example and has no useful application then it should probably not be in the library.

Reply:
I agree with this. It was useful at the beginning when  I was looking into how to design class `Application` supported by a convenient command-line parser. And now this test could be safely removed. I will do it now.